### PR TITLE
feat(dli): the flink job resources supports updating tags

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

The flink job resources supports updating tags.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. the flink job resources supports tags.
2. modifying related document and acceptance test.

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccResourceDliFlinkJob_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccResourceDliFlinkJob_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceDliFlinkJob_basic
=== PAUSE TestAccResourceDliFlinkJob_basic
=== CONT  TestAccResourceDliFlinkJob_basic
--- PASS: TestAccResourceDliFlinkJob_basic (473.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       473.090s

make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccResourceDliFlinkJarJob_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccResourceDliFlinkJarJob_ -timeout 360m -parallel 4
=== RUN   TestAccResourceDliFlinkJarJob_basic
=== PAUSE TestAccResourceDliFlinkJarJob_basic
=== RUN   TestAccResourceDliFlinkJarJob_all
=== PAUSE TestAccResourceDliFlinkJarJob_all
=== CONT  TestAccResourceDliFlinkJarJob_basic
=== CONT  TestAccResourceDliFlinkJarJob_all
--- PASS: TestAccResourceDliFlinkJarJob_all (355.42s)
--- PASS: TestAccResourceDliFlinkJarJob_basic (391.09s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       391.132s
```

### CODE REVIEW

File: docs/resources/dli_flinkjar_job.md [hunk 1]
```diff
@@ -125,7 +125,7 @@ The following arguments are supported:
 
 * `runtime_config` - (Optional, Map) Specifies customizes optimization parameters when a Flink job is running.
 
-* `tags` - (Optional, Map, ForceNew) Specifies the key/value pairs to associate with the resource.
+* `tags` - (Optional, Map) Specifies the key/value pairs to associate with the resource.
  Changing this parameter will create a new resource.
```
Comment (Lance52259) [inline docs/resources/dli_flinkjar_job.md]:
```suggestion
```
forgot to remove this content

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_flinkjar_job_test.go [hunk 2]
```diff
@@ -62,57 +71,86 @@ func TestAccResourceDliFlinkJarJob_basic(t *testing.T) {
...
-  name            = "%s"
+  name            = "%[2]s"
+  queue_name      = "common_test"
+  entrypoint      = "${huaweicloud_dli_package.test.group_name}/${huaweicloud_dli_package.test.object_name}"
+  entrypoint_args = "--output.path obs://${var.ak}:${var.sk}@obs.%[3]s.myhuaweicloud.com/${huaweicloud_obs_bucket.test.bucket}/output"
+  obs_bucket      = huaweicloud_obs_bucket.test.bucket
+  log_enabled     = true
+
+  tags = {
+	foo = "bar"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_flinkjar_job_test.go]: wrong format, please check this and the other places

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinksql_job.go [hunk 3]
```diff
@@ -516,6 +574,8 @@ func updateFlinkSqlJobWithStop(ctx context.Context, client *golangsdk.ServiceCli
 	d *schema.ResourceData) diag.Diagnostics {
 
 	if d.HasChangesExcept("smn_topic", "restart_when_exception", "resume_checkpoint", "resume_max_num", "obs_bucket") {
+		log.Print(3333)
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_flinksql_job.go]: This print thing is using for local debug?

File: docs/resources/dli_flinkjar_job.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/resources/dli_flinkjar_job.md]: fixed

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_flinkjar_job_test.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_flinkjar_job_test.go]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinksql_job.go (hunk 3 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_flinksql_job.go]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go [hunk 4]
```diff
@@ -27,6 +27,10 @@ import (
 // @API DLI PUT /v1.0/{project_id}/streaming/flink-jobs/{job_id}
 // @API DLI POST /v1.0/{project_id}/streaming/jobs/stop
 // @API DLI DELETE /v1.0/{project_id}/streaming/jobs/{job_id}
+// @API DLI DELETE /v1.0/{project_id}/streaming/jobs/{job_id}
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go]:
```suggestion
// @API DLI DELETE /v1.0/{project_id}/streaming/jobs/{job_id}
```
same reference for the delete API?

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go [hunk 5]
```diff
@@ -366,34 +377,40 @@ func resourceFlinkJarJobDelete(_ context.Context, d *schema.ResourceData, meta i
...
+			return diag.Errorf("error creating DLI v1 client, err=%s", err)
+		}
 
-	diagErr := authorizeObsBucket(client, d)
-	if diagErr != nil {
-		return diagErr
-	}
+		jobId, iErr := strconv.Atoi(d.Id())
+		if iErr != nil {
+			return diag.Errorf("the DLI flink job_id must be number. actual id=%s", d.Id())
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go]:
```suggestion
			return diag.Errorf("the DLI flink job_id must be number, but the actual ID is '%s'", d.Id())
```
SQL job resource ditto

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go (hunk 4 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go]: fixed

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go (hunk 5 above)
Comment (wuzhuanhong) [reply huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go]: fixed

**Pull request merged**